### PR TITLE
Implement new capabilities (persistence, ttl, encryption and queryable)

### PR DESCRIPTION
**Pull request by mmandlis**
the new capabilities aren't used yet.
the next steps would be to:
- reimplement capabilities resolver
- then in KT
- then migrate all usage (manifest should remain unchanged and this migration will be backward compatible).

### CODE REVIEW

File: src/runtime/capabilities-new.ts [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+  isQueryable(): boolean {
+    const queryable = this.list.find(capability => capability instanceof Queryable);
+    return queryable ? (queryable as Queryable).value : false;
+  }
+
+  compare(other: Capability): CapabilityComparison {
+    assert(other instanceof Capabilities);
+    const otherCapabilities = other as Capabilities;
+    assert(this.list.length === otherCapabilities.list.length);
+    for (let index = 0; index < this.list.length; ++ index) {
```
Comment (csilvestrini) [inline src/runtime/capabilities-new.ts:327]: Nit: `++index`

File: src/runtime/capabilities-new.ts [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+    }
+    return CapabilityComparison.Equivalent;
+  }
+
+  // Returns true, if all capabilities in one sets are consistently equivalent
+  // or stricter than the other.
+  protected canRestrict(other: Capabilities): boolean {
+    assert(this.list.length === other.list.length);
+    let totalComparison = CapabilityComparison.Equivalent;
+    for (let index = 0; index < this.list.length; ++ index) {
```
Comment (csilvestrini) [inline src/runtime/capabilities-new.ts:343]: Here too

File: src/runtime/capabilities-new.ts (hunk 1 above)
Comment (mmandlis) [reply src/runtime/capabilities-new.ts]: done

File: src/runtime/capabilities-new.ts (hunk 2 above)
Comment (mmandlis) [reply src/runtime/capabilities-new.ts]: done

**Comment by mmandlis** [comment]
manual copybara succeeded in cl/314773766, merging now.

**Pull request merged**